plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/IcebergTableProcedureId.java

```diff
@@ -15,5 +15,17 @@
 
 public enum IcebergTableProcedureId
 {
-    OPTIMIZE,
+    OPTIMIZE(true);
+
+    private boolean readsData;
+
+    private IcebergTableProcedureId(boolean readsData)
+    {
+        this.readsData = readsData;
+    }
+
+    public boolean isReadsData()
```

Comment:
a "procedure id" shouldn't have `isReadsData` attribute. We would need to rename the class.

For now, let's add this as an ordinary field in `IcebergTableExecuteHandle`